doc/developer/design/20230225_compute_introspection_schema.md

```diff
@@ -0,0 +1,232 @@
+nanoseconds, whereas their corresponding non-`raw` views convert the
+times to the SQL interval type.
+
+Confusingly, not all times are reported as intervals; for example,
+`mz_scheduling_parks` has `slept_for` and `requested` columns, whose
+type is `bigint` and whose unit is undocumented (in fact, it is nanoseconds).
+
+The trade-off between intervals and nanoseconds timestamps is that the
+former is more ergonomic but also has only microsecond precision.
+
+#### Inconsistent presence of `sum` column in histograms
+
+Some histograms include a per-bucket sum, which allows consumers to
+compute average or total values precisely. Confusingly, not all
+histograms report such a column (e.g. `mz_worker_compute_delays` does
+not, whereas `mz_worker_raw_compute_delays` does).
+
+We introduced the `sum` column initially to support Prometheus
+histograms, which expects to know about cumulative sums and cannot be
+convinced otherwise. However, it is possible to provide approximate
+sums, based on multiplying the bucket values with the bucket counts.
+
+#### Lack of clarity around per-replica vs. global relations
+
+Many of the relations are logging sources that run locally in
+clusters, are not persisted, and report different data depending on
+the currently selected replica. Views based on these sources are also
+per-replica. It is not straightforward to tell from our docs which is
+the case for each relation.
+
+Furthermore, sometimes the same logical value is reported by two
+relations, one reporting `clusterd`'s view of the world, and the other
+reporting `environmentd`'s. For example, `mz_compute_frontiers`
+vs. `mz_cluster_replica_frontiers`. However, the names do not make it
+clear what the distinction is. 
+
+### Suggested fixes for schema inconsistencies
+
+* All per-worker relations should have a corresponding "global" view
+  where the results are aggregated across workers in the way most
+  likely to be useful (for example, summing, taking the minimum, or
+  restricting to worker 0). Because these global views are more likely
+  to be useful than the underlying sources, they should have the basic
+  name (e.g., `mz_arrangement_sharing`). The underlying source should
+  have a name with `per_worker` suffixed (e.g.,
+  `mz_arrangement_sharing_per_worker`). Furthermore, the per-worker
+  variant should be clearly less prominent in documentation (either at
+  the bottom of the page, or in a "see more" section that has to be
+  clicked to expand).
+* The expectations about per-worker relations should be clearly
+  documented (e.g., we should clearly say whether the values for each
+  worker are expected to be the same during normal operation).
+* All relations that report a simple count or sum should be views based on
+  an `_internal` variant that maintains the count or sum in the
+  Differential diff field. We may consider renaming this, since it
+  conflicts confusingly with the overall name `mz_internal`.
```

Comment:
How about using a `_raw` suffix instead of `_internal`? That is free of ambiguity once we remove the `mz_raw_*` variants. Also reflects quite well that the data in these collections consists of the raw un-aggregated DD updates.